db/hybrid_logical_vector_test.go

```diff
@@ -358,3 +358,153 @@ func (h *HLVAgent) insertWithHLV(ctx context.Context, key string) (casOut uint64
 	require.NoError(h.t, err)
 	return cas
 }
+
+// TestExtractHLVFromChangesMessage:
+//   - Test case 1: CV entry and 1 PV entry
+//   - Test case 2: CV entry and 2 PV entries
+//   - Test case 3: CV entry, 2 MV entries and 2 PV entries
+//   - Test case 4: just CV entry
+//   - Each test case gets run through extractHLVFromBlipMessage and assert that the resulting HLV
+//     is correct to what is expected
+func TestExtractHLVFromChangesMessage(t *testing.T) {
+	testCases := []struct {
+		name             string
+		hlvString        string
+		expectedHLV      []string
+		mergeVersions    bool
+		previousVersions bool
+	}{
+		{
+			name:             "test case 1",
+			hlvString:        "1@Hell0CA; 1@1Hr0k43xS662TToxODDAxQ",
+			expectedHLV:      []string{"Hell0CA@1", "1Hr0k43xS662TToxODDAxQ@1"},
+			previousVersions: true,
+		},
+		{
+			name:             "test case 2",
+			hlvString:        "25@def; 20@abc,18@hij",
+			expectedHLV:      []string{"def@25", "abc@20", "hij@18"},
+			previousVersions: true,
+		},
+		{
+			name:             "test case 3",
+			hlvString:        "25@def; 22@def,21@eff; 20@abc,18@hij",
+			expectedHLV:      []string{"def@25", "abc@20", "hij@18", "m_def@22", "m_eff@21"},
```

Comment:
Does this mean the string representation of a HLV is misaligned between CBL and SG/XDCR?

E.g. `25@def` == `def@25`

That seems like a major annoyance and possibly has performance implications (we'd need to parse/reformat rather than passing things straight through)

Reply:
So I can change this if you think it will be easier? The expected HLV list you see there is input for a test only function i created a while back (called createHLVForTest), rather annoyingly created before I read any CBL replication docs and I opted for the reverse order in the format when creating this function.  

Should I align it with CBL's representation?

Reply:
Ahh I see. I think we should reverse/change it, if this is just our test-representation.